File: Frontend/src/api/models/programHistoryDTO.ts
import type { HistorySourceType } from "./historySourceType";
import type { ProgramStatuses } from "./programStatuses";

export interface ProgramHistoryDTO {
  userid: string;
  userFIO: string;
  programId: string;
  date: string;
  oldStatus: ProgramStatuses;
  newStatus: ProgramStatuses;
  sourceId: string;
  sourceType: HistorySourceType;
}

File: Backend/Application/DTO/Programs/ProgramHistoryDTO.cs
using Domain.Entities.ProgramContext;
using Domain.Entities.References;
using Domain.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace Application.DTO.Programs
{
    public class ProgramHistoryDTO
    {
        public Guid UserId { get; set; }
        public string UserFIO { get; set; }
        public Guid ProgramId { get; set; }
        public DateTimeOffset Date { get; set; }
        public ProgramStatuses OldStatus { get; set; }
        public ProgramStatuses NewStatus { get; set; }
        public Guid SourceId { get; set; }
        public HistorySourceType SourceType { get; set; }
    };
}
